apps/common/lib/lexical/ast.ex

```diff
@@ -88,6 +88,44 @@ defmodule Lexical.Ast do
   @type short_alias :: atom()
   @type alias_segments :: [short_alias]
 
+  @doc """
+  Analyzes a document.
+  """
+  @spec analyze(Document.t()) :: Analysis.t()
+  def analyze(%Document{} = document) do
+    document
+    |> from()
+    |> Analysis.new(document)
+  end
+
+  @doc """
+  Attempts to analyze a document at least until `position`.
+
+  This can be used to analyze an invalid AST up until, for instance, the
+  cursor position. The full analysis will be completed if the AST is
+  valid.
+
+  This function is not guaranteed to return a valid analysis.
```

Comment:
then why is it called safe? 🤔 

(i get it, because it can't crash)

Reply:
I don't love the name, but I wanted *something* that indicated that it's *trying some stuff* -- I don't think "safe" is the right word.